### PR TITLE
[CI] Fix Windows Wheel flaky build caused by ', using heredoc in refreshenv.

### LINKED ISSUE #29006: [CI] `linux://python/ray/dashboard:modules/job/tests/test_sdk` is failing/flaky on master.
- 7d5df19275519e17e0cda05d551a1da934fdf286 FAILED [Buildkite :serverless: Dashboard Tests](https://buildkite.com/ray-project/oss-ci-build-branch/builds/305#01839c2c-31da-4fc3-9b30-ebde3e294e85)
- f68d82b714d8497053cda0ca8e3732e37a103214 FAILED [Buildkite :serverless: Dashboard Tests](https://buildkite.com/ray-project/oss-ci-build-branch/builds/301#01839563-63a7-414b-b852-4e9a225ef244)
- 5ce0efaac9ea287d841103efbe173e9457ec442e FAILED [Buildkite :serverless: Dashboard Tests](https://buildkite.com/ray-project/oss-ci-build-branch/builds/296#018391ff-6bde-4085-9405-d1f9e3f24f90)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-linux://python/ray/dashboard:modules/job/tests/test_sdk-END
....

### LINKED ISSUE #29717: [CI] `bk://:windows: Wheels` is failing/flaky on master.
- 98da294f0b8fbf0ab9d94ebd552c4d78b4d3aaf8 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/737#01841506-8234-4946-a16e-4e72e73f8568)
- 875193ca8d4720a7215aad2a275851f5157674c2 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/733#018412b7-333c-4a6d-9698-184d8e584fa5)
- 161567dbd93f94bef941c17779d3880f310a0373 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/730#01841271-2261-4592-ade3-76ce92d785a2)
- 8b4283fb3da524c63a41f34a5221e762a3dc1587 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/726#0184111b-fc6f-40d7-9f6b-9498065cc345)
- bf22325eb518c4a42242a88031909869da003850 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/723#0184108b-7f56-40ac-9d2f-cc42dc7f90bf)
- 51a7b093c9fd56d4b7a14d4159fe6a6bf770742e FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/722#01841064-ba08-4701-ad46-dfb69d5ac7b2)
- e464bf07af9f6513bf71156d1226885dde7a8f46 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/718#01841036-570e-482b-bcfb-bc18dc646673)
- 2ca05c84d0eca54ea2cfd438577afe40be796cd8 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/716#0184100c-47c3-4f89-a68a-ff465258b647)
- 8e3e741d9b94a2694e789d4bcdff16e4c27a501c FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/708#01840f26-2a99-40f7-abbb-c926e5dfb0d6)
- 9a3c0fb63a08590b7ee888665506b78b0a07d7d1 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/695#01840c2b-aec2-43d2-8165-463b98f5d993)
- d4ea9fbf0a84f668a2f3935f9efc3386dd08691d FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/690#01840c0b-b910-4fc4-a9b8-a9924dda7248)
- 2cbeca2393d56a682fdd4bc1b0042bfad11f4e93 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/688#01840bc6-1b7b-43e6-b418-2d55c368ae18)
- f8d89c69e8b8ce9caca6361c2b7cc94e88011cfd FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/681#01840ace-203e-4744-a0c9-a96a06e17c32)
- de238dd626a48a16c8b3cd006f3482db75f63a83 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/673#01840260-d982-4aa8-b6ec-34a1f9089fe5)
- 39a730c21ac51e7f6d66d775a48f3bf122243411 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/666#0183fc67-ef06-4487-b546-6e1a3759679d)
- ef9649c1c220890c04c29e0b224fbaaca2c7c038 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/656#0183fb22-9ae9-4e9d-842a-55f306f11a9b)
- f85e555dfbb0ba502e0d5f765a2f5fe15bcd55b6 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/647#0183f93d-e995-43cc-b6a9-20a10c611686)
- 470eac627b1a6473afaafd75f98ab35f1f142857 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/643#0183f880-19ff-4551-a2f5-b3dd2d0b9ac5)
- 2c032ebe0ce8e625796491030a759e5e01edbef9 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/636#0183f70a-b7a0-486f-9885-ee8196c98d83)
- a04687c2e7a6b2063bd83823d4985255e7fced83 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/621#0183f4ef-8f03-4a64-a5cd-f3836d48e1e9)
- 0de2f18c2529cc98245e1a8895a894b35fdde7c7 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/610#0183f279-32aa-40ce-8be8-0690991ee49b)
- 4bcfb80ad0205ec08ef8e10a3678f69fe36ae866 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/606#0183f224-0412-42a9-b018-de0b487af6a4)
- 72c570ecef96fb6cc43252fbbb06cc03d3b58f93 FAILED [:windows: Wheels](https://buildkite.com/ray-project/oss-ci-build-branch/builds/605#0183f1fb-52d0-4ba4-aca9-ce3434661bea)

....
Generated from flaky test tracker. Please do not edit the signature in this section.
DataCaseName-bk://:windows: Wheels-END
....

### PR DESCRIPTION
**Pull request by cadedaniel**
Screen Shot 2022-11-01 at 12 20 06 PM

### Description
See https://github.com/ray-project/ray/pull/29878 and https://github.com/ray-project/ray/issues/29717 for context.

TL;DR there is a script which exports environment variables present in powershell to bash, but it fails to handle cases where environment variable values contain single quotes. This causes the Windows Wheels build to fail whenever an environment variable value contains a single quote, particularly when the commit message has a single quote (common!). I fixed it by exporting the variables using bash heredocs which are more robust against arbitrary input values.

### Testing
Powershell code to test:
```pwsh
#Import-Module "$env:ChocolateyInstall\helpers\chocolateyProfile.psm1"
#Update-SessionEnvironment
# Round brackets in variable names cause problems with bash
Get-ChildItem env:* | %{
  if (!($_.Name.Contains('('))) {
    $value = $_.Value
    if ($_.Name -eq 'PATH') {
      $value = $value -replace ';',':'
    }
    # Write-Output ("export " + $_.Name + "='" + $value + "'")
    Write-Output ("export " + $_.Name + "=$`(cat <<- 'REFRESHENV_EOF'`n" + $value + "`nREFRESHENV_EOF`)")
  }
}
```

Test runner (COMPLEX_ENV_VAR_1 has single quotes):
```bash
#!/usr/bin/env bash

pwsh=$(which pwsh)

export COMPLEX_ENV_VAR_1=$(cat <<- "EOF"
[Jobs] Deflake jobs sdk test (#29707)
--
Wait for all nodes to come up before running the unit test to deflake the unit test. It submits a bunch of jobs and checks that at least one was run on a worker node. However if the jobs are submitted too soon, its' possible that only the head node has come up, so 'theyll all be scheduled on the head node. The fix is to only submit the jobs after all the nodes are up.
Also adds some logs to improve debuggability.

Related issue number
Closes #29006
EOF)

env -i COMPLEX_ENV_VAR_1="$COMPLEX_ENV_VAR_1" $pwsh ps.ps | sed 's/export PATH/export FAKE_PATH/g' > refreshenv.sh
echo 'env' >> refreshenv.sh

env -i bash refreshenv.sh
```

Output:
```
$ ./test.sh
FAKE_PATH=/usr/local/microsoft/powershell/7:
COMPLEX_ENV_VAR_1=[Jobs] Deflake jobs sdk test (#29707)
--
Wait for all nodes to come up before running the unit test to deflake the unit test. It submits a bunch of jobs and checks that at least one was run on a worker node. However if the jobs are submitted too soon, its' possible that only the head node has come up, so 'theyll all be scheduled on the head node. The fix is to only submit the jobs after all the nodes are up.
Also adds some logs to improve debuggability.

Related issue number
Closes #29006
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
PSModulePath=/tmp/71636ff1-42ce-40a1-9c08-d4e9ae135fad/.local/share/powershell/Modules:/usr/local/share/powershell/Modules:/usr/local/microsoft/powershell/7/Modules
PWD=/Users/cade/dev/cadelabs/ray-powershell-fix
SHLVL=1
_=/usr/bin/env
```

It worked! Without the `heredoc`, it gets borked:
```
$ ./test.sh
refreshenv.sh: line 3: export: `up,': not a valid identifier
refreshenv.sh: line 3: export: `theyll all be scheduled on the head node. The fix is to only submit the jobs after all the nodes are up.
Also adds some logs to improve debuggability.

Related issue number
Closes #29006': not a valid identifier
FAKE_PATH=/usr/local/microsoft/powershell/7:
COMPLEX_ENV_VAR_1=[Jobs] Deflake jobs sdk test (#29707)
--
Wait for all nodes to come up before running the unit test to deflake the unit test. It submits a bunch of jobs and checks that at least one was run on a worker node. However if the jobs are submitted too soon, its
__CF_USER_TEXT_ENCODING=0x1F5:0x0:0x0
PSModulePath=/tmp/81c31618-cb53-4d44-b01c-fd622dba6734/.local/share/powershell/Modules:/usr/local/share/powershell/Modules:/usr/local/microsoft/powershell/7/Modules
PWD=/Users/cade/dev/cadelabs/ray-powershell-fix
SHLVL=1
_=/usr/bin/env
```

### CODE REVIEW

**Review by jjyao — APPROVED** [review]
Could you add a single quote to the commit message to make sure it works :)

**Comment by cadedaniel** [comment]
haha, sure :)

**Comment by cadedaniel** [comment]
@jjyao merge please

**Comment by cadedaniel** [comment]
🤔🤔🤔🤔
```
+ source /c/Users/ContainerAdministrator/AppData/Local/Temp/refreshenv.sh
/c/Users/ContainerAdministrator/AppData/Local/Temp/refreshenv.sh: line 6: warning: here-document at line 4 delimited by end-of-file (wanted `REFRESHENV_EOF')
+++ cat
++ export BUILDKITE=true
++ BUILDKITE=true
/c/Users/ContainerAdministrator/AppData/Local/Temp/refreshenv.sh: line 9: warning: here-document at line 7 delimited by end-of-file (wanted `REFRESHENV_EOF')
+++ cat
++ export 'ProgramW6432=C:\Program Files'
++ ProgramW6432='C:\Program Files'
/c/Users/ContainerAdministrator/AppData/Local/Temp/refreshenv.sh: line 12: warning: here-document at line 10 delimited by end-of-file (wanted `REFRESHENV_EOF')
+++ cat
++ export 'ChocolateyInstall=C:\ProgramData\chocolatey'
++ ChocolateyInstall='C:\ProgramData\chocolatey'
/c/Users/ContainerAdministrator/AppData/Local/Temp/refreshenv.sh: line 15: warning: here-document at line 13 delimited by end-of-file (wanted `REFRESHENV_EOF')
```
